src/script/sign.cpp

```diff
@@ -629,6 +679,18 @@ bool PSBTInput::IsSane() const
     return true;
 }
 
+bool PSBTInput::GetUTXO(CTxOut& utxo, int prevout_index) const
```

Comment:
This interface feels fragile to me. Could this instead be "bool PartiallySignedTransaction::GetInputUTXO(CTxOut& utxo, int input_index)"? That version would be resistant to misuse, whereas this version will do non-obvious crazy things if you accidentally give it the wrong prevout_index for your input (which seems easy to do.)

Reply:
Done